config/v1/types_infrastructure.go

```diff
@@ -212,11 +214,14 @@ type PlatformStatus struct {
 // AWSServiceEndpoint store the configuration of a custom url to
 // override existing defaults of AWS Services.
 type AWSServiceEndpoint struct {
-	// Name is the name of the AWS service.
+	// name is the name of the AWS service.
+	// The list of all the service names can be found at https://docs.aws.amazon.com/general/latest/gr/aws-service-information.html
 	// This must be provided and cannot be empty.
+	//
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
 	Name string `json:"name"`
 
-	// URL is hostname only or fully qualified URI that overrides the default generated
+	// url is hostname only or fully qualified URI with schemes https or http, that overrides the default generated
```

Comment:
What requires HTTP?  Do we really want to allow that.  If we don't know of anything, let's require secure to match the posture of the rest of the product.

Reply:
> What requires HTTP? Do we really want to allow that. If we don't know of anything, let's require secure to match the posture of the rest of the product.

AWS SDK supports it, and that's way I have it here, keeping the parity with AWS SDK.

Reply:
dropped it in https://github.com/openshift/api/pull/599/commits/1bd3edd644f0585153452d0b901ee3e811f7a19d